projects/core/src/cart/facade/active-cart.service.ts

```diff
@@ -0,0 +1,242 @@
+import { Injectable } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { Observable, of } from 'rxjs';
+import {
+  filter,
+  map,
+  shareReplay,
+  take,
+  tap,
+  switchMap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { AuthService } from '../../auth/index';
+import { Cart } from '../../model/cart.model';
+import { OrderEntry } from '../../model/order.model';
+import { CartActions } from '../store/actions/index';
+import { MultiCartSelectors } from '../store/selectors/index';
+import { StateWithMultiCart } from '../store/multi-cart-state';
+import { LowLevelCartService } from './low-level-cart.service';
+import { LoaderState } from '../../state/utils/loader/loader-state';
+import {
+  OCC_USER_ID_CURRENT,
+  OCC_USER_ID_ANONYMOUS,
+} from '../../occ/utils/occ-constants';
+import { getCartIdByUserId } from '../utils/utils';
+
+// TODO document methods
+
+@Injectable()
+export class ActiveCartService {
+  private readonly PREVIOUS_USER_ID_INITIAL_VALUE =
+    'PREVIOUS_USER_ID_INITIAL_VALUE';
+  private previousUserId = this.PREVIOUS_USER_ID_INITIAL_VALUE;
+  private _activeCart$: Observable<Cart>;
+
+  private userId = OCC_USER_ID_ANONYMOUS;
+  private cartId;
+
+  private activeCartId = this.store.pipe(
```

Comment:
Bagatelle: For readability (consistency), can we mark activeCartId and cartSelector with `$` as observables?